pkg/container-collection/options.go

```diff
@@ -590,42 +591,33 @@ func WithOCIConfigEnrichment() ContainerCollectionOption {
 				return true
 			}
 
-			var containerTypeAnnotation, containerNameAnnotation, podNameAnnotation, podNamespaceAnnotation, podUIDAnnotation string
-			if cm := container.OciConfig.Annotations[crioContainerManagerAnnotation]; cm == runtimeclient.CrioName {
-				container.Runtime = runtimeclient.CrioName
-				containerTypeAnnotation = crioContainerTypeAnnotation
-				containerNameAnnotation = crioContainerNameAnnotation
-				podNameAnnotation = crioPodNameAnnotation
-				podNamespaceAnnotation = crioPodNamespaceAnnotation
-				podUIDAnnotation = crioPodUIDAnnotation
-			} else if _, isContainerd := container.OciConfig.Annotations[containerdContainerTypeAnnotation]; isContainerd {
-				container.Runtime = runtimeclient.ContainerdName
-				containerTypeAnnotation = containerdContainerTypeAnnotation
-				containerNameAnnotation = containerdContainerNameAnnotation
-				podNameAnnotation = containerdPodNameAnnotation
-				podNamespaceAnnotation = containerdPodNamespaceAnnotation
-				podUIDAnnotation = containerdPodUIDAnnotation
-			} else {
+			resolver, err := ociannotations.NewResolver(container.OciConfig.Annotations)
+			if err != nil {
+				log.Warnf("OCIConfig enricher: failed to initialize annotation resolver")
```

Comment:
Isn't the resolver expected to fail when using docker containers instead of containerd/cri-o?

Reply:
Yes, it should. Mainly added this for debugging purpose. Should I change it to `log.Debugf` ?